source/common/config/config_provider_impl.h

```diff
@@ -49,27 +50,36 @@ namespace Config {
+//      be called to fetch either an existing ConfigSubscriptionCommonBase if the config
+//      source configuration matches, or a newly instantiated subscription otherwise.
 //
 // For immutable providers:
-//   1) Create a class derived from ImmutableConfigProviderImplBase and implement the required
+//   1) Create a class derived from ImmutableConfigProviderBase and implement the required
 //   interface.
 //
 // For mutable (xDS) providers:
-//   1) Create a class derived from MutableConfigProviderImplBase and implement the required
-//   interface.
-//   2) Create a class derived from ConfigSubscriptionInstanceBase; this is the entity
-//   responsible for owning and managing the Envoy::Config::Subscription<ConfigProto> that provides
-//   the underlying config subscription.
+//   1) According to the API type, create a class derived from MutableConfigProviderBase or
+//   DeltaMutableConfigProviderBase and implement the required interface.
+//   2) According to the API type, create a class derived from ConfigSubscriptionInstance or
+//   DeltaConfigSubscriptionInstance; this is the entity responsible for owning and managing the
+//   Envoy::Config::Subscription<ConfigProto> that provides the underlying config subscription.
+//     a) For a ConfigProvider::ApiType::Full subscription instance (i.e., a
+//     ConfigSubscriptionInstance child):
 //     - When subscription callbacks (onConfigUpdate, onConfigUpdateFailed) are issued by the
-//     underlying subscription, the corresponding ConfigSubscriptionInstanceBase functions must be
-//     called as well.
-//     - On a successful config update, checkAndApplyConfig() should be called to instantiate the
-//     new config implementation and propagate it to the shared config providers and all
-//     worker threads.
-//       - On a successful return from checkAndApplyConfig(), the config proto must be latched into
-//       this class and returned via the getConfigProto() override.
+//     underlying subscription, the corresponding ConfigSubscriptionInstance functions
+//     must be called as well.
+//     - On a successful config update, checkAndApplyConfigUpdate() should be called to instantiate
+//     the new config implementation and propagate it to the shared config providers and all worker
+//     threads.
+//       - On a successful return from checkAndApplyConfigUpdate(), the config proto must be latched
+//       into this class and returned via the getConfigProto() override.
+//    b) For a ConfigProvider::ApiType::Delta subscription instance (i.e., a
+//    DeltaConfigSubscriptionInstance child):
+//    - When subscription callbacks (onConfigUpdate, onConfigUpdateFailed) are issued by the
+//    underlying subscription, the corresponding ConfigSubscriptionInstance functions must be called
+//    as well.
+//    - On a successful config update, applyConfigUpdate() should be called to propagate the config
+//    updates to all bound config providers and worker threads.
```

Comment:
How is `getConfigProto()` supported in this world?

Reply:
Updating documentation to clarify this.